src/server/scripts/Outland/CoilfangReservoir/TheUnderbog/boss_swamplord_muselek.cpp

```diff
@@ -0,0 +1,398 @@
+        me->SetReactState(REACT_AGGRESSIVE);
+
+        events.ScheduleEvent(EVENT_SHOOT, 0s);
+        events.ScheduleEvent(EVENT_MULTI_SHOT, 20s, 30s);
+
+        events.ScheduleEvent(EVENT_KNOCK_AWAY, 35s, 40s);
+        events.ScheduleEvent(EVENT_BEAR_COMMAND, 10s);
+
+        events.ScheduleEvent(EVENT_COMBAT_SEQUENCE_1, 15s, 20s);
+    }
+
+    void AttackStart(Unit* who) override
+    {
+        ScriptedAI::AttackStartCaster(who, 30.0f);
+    }
+
+    void OnSpellCast(SpellInfo const* spell) override
+    {
+        if (spell->Id == SPELL_BEAR_COMMAND)
+            Talk(SAY_COMMAND);
+    }
+
+    void MovementInform(uint32 type, uint32 pointId) override
+    {
+        if (type != POINT_MOTION_TYPE)
+            return;
+
+        if (pointId == POINT_MOVE_AWAY)
+            events.ScheduleEvent(EVENT_COMBAT_SEQUENCE_3, 1s);
+    }
+
+    void KilledUnit(Unit* /*victim*/) override
+    {
+        Talk(SAY_SLAY);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        Talk(SAY_DEATH);
+        _JustDied();
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
+
+        events.Update(diff);
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
+
+        while (uint32 eventId = events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_SHOOT:
+                    DoCastVictim(SPELL_SHOOT);
+                    events.Repeat(2s, 4s);
+                    break;
+                case EVENT_MULTI_SHOT:
+                    DoCastVictim(SPELL_MULTI_SHOT);
+                    events.Repeat(25s, 35s);
+                    break;
+                case EVENT_KNOCK_AWAY:
+                    DoCastVictim(SPELL_KNOCK_AWAY);
+                    events.Repeat(30s, 40s);
+                    break;
+                case EVENT_BEAR_COMMAND:
+                    DoCastSelf(SPELL_BEAR_COMMAND);
+                    events.Repeat(25s);
+                    break;
+
+                // This part requires additional research
+                case EVENT_COMBAT_SEQUENCE_1:
+                    DoCastSelf(SPELL_HUNTERS_MARK);
+
+                    if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
+                        DoCast(target, SPELL_FREEZING_TRAP);
+
+                    me->SetReactState(REACT_PASSIVE);
+                    events.ScheduleEvent(EVENT_COMBAT_SEQUENCE_2, 2s);
+                    events.Repeat(25s, 50s);
+                    break;
+                case EVENT_COMBAT_SEQUENCE_2:
+                    DoCastSelf(SPELL_PACIFY_SELF);
+                    DoCastSelf(SPELL_MOVE_AWAY_PRIMER);
+
+                    // This is based only on one cast in sniff, need more info
+                    if (roll_chance_i(50))
+                        DoCastSelf(SPELL_DETERRENCE);
+                    break;
+                case EVENT_COMBAT_SEQUENCE_3:
+                    // NYI
```

Comment:
So this PR is not ready?

Reply:
It is ready, I'm not going to implement this spell as I don't really understand how it works